starknet-providers/src/jsonrpc/mod.rs

```diff
@@ -301,18 +307,15 @@ where
     }
 
     /// Estimate the fee for a given StarkNet transaction
-    pub async fn estimate_fee<R>(
+    pub async fn estimate_fee(
         &self,
-        request: R,
+        request: &BroadcastedTransaction,
         block_id: &BlockId,
-    ) -> Result<FeeEstimate, JsonRpcClientError<T::Error>>
-    where
-        R: AsRef<FunctionCall>,
-    {
+    ) -> Result<FeeEstimate, JsonRpcClientError<T::Error>> {
         self.send_request(
```

Comment:
The use of generic here is by design. In some cases it's impossible to pass references since the value gets dropped immediately. Using `AsRef` generic here allows the function to be used in those cases.

Extra caution is needed if you're changing the design.

Reply:
Reverted to using the generic for estimate_fee